game-core/src/main/java/games/strategy/triplea/attachments/UnitAttachment.java

```diff
@@ -2568,14 +2570,26 @@ public void validate(final GameData data) throws GameParseException {
   }
 
   private Collection<Territory> getListedTerritories(final String[] list) throws GameParseException {
-    final List<Territory> territories = new ArrayList<>();
+    final Set<Territory> territories = new HashSet<>();
     for (final String name : list) {
       // Validate all territories exist
       final Territory territory = getData().getMap().getTerritory(name);
-      if (territory == null) {
-        throw new GameParseException("No territory called: " + name + thisErrorMsg());
+      if (territory != null) {
+        territories.add(territory);
+      } else {
+        // Check if its a territory effect and get all territories
+        if (getData().getTerritoryEffectList().keySet().contains(name)) {
```

Comment:
I believe the map interface defines a containsKey method, so it's uneccessary to invoke keySet()

Reply:
Updated.